### PR TITLE
Fixes #8553: Use the list-compatible-input equivalent on Windows system

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/8553

### CODE REVIEW

File: techniques/system/common/1.0/promises.st [hunk 1]
```diff
@@ -124,21 +127,20 @@ bundle common va
...
+      # We cannot merge both calls, as the paths have a space in them, and we need to quote them on windows
+      "ncf_common_inputs_10" slist => splitstring(execresult("${list_compatible_inputs} ${sys.cf_version} '${ncf_path}/common' 10_ncf_internals", "powershell"), "\n", 10000);
+      "ncf_common_inputs_20" slist => splitstring(execresult("${list_compatible_inputs} ${sys.cf_version} '${ncf_path}/common' 20_cfe_basics", "powershell"), "\n", 10000);
+      "ncf_common_inputs_30" slist => splitstring(execresult("${list_compatible_inputs} ${sys.cf_version} '${ncf_path}/common' 30_generic_methods", "powershell"), "\n", 10000);
+      "ncf_common_inputs_40" slist => splitstring(execresult("${list_compatible_inputs} ${sys.cf_version} '${ncf_path}/common' 40_it_ops_knowledge", "powershell"), "\n", 10000);
+      "ncf_common_inputs_50" slist => splitstring(execresult("${list_compatible_inputs} ${sys.cf_version} '${ncf_path}/common' 50_techniques", "powershell"), "\n", 10000);
```
Comment (ncharles) [inline techniques/system/common/1.0/promises.st]: caution during upmerge, in branch 3.2, there are no 50 included there anymore

File: techniques/system/common/1.0/promises.st [hunk 2]
```diff
@@ -99,14 +99,17 @@ bundle common va
...
+      "list_compatible_inputs" string => "NCF_CACHE_PATH=${sys.workdir}/state /bin/sh ${ncf_path}/common/10_ncf_internals/list-compatible-inputs";
+
     windows::
       "rudder_base"    string => "${sys.winprogdir}\Rudder";
       "rudder_tools"   string => "${rudder_base}\sbin";
-      "ncf_path"       string => "${rudder_base}\ncf";
+      "ncf_path"       string => "${rudder_base}\var\ncf";
 
-    !windows::
+      # & is for powershell commands
```
Comment (peckpeck) [inline techniques/system/common/1.0/promises.st]: even in a comment stringtemplate won't like the &

**Comment by ncharles** [comment]
Commit modified

File: techniques/system/common/1.0/promises.st [hunk 3]
```diff
@@ -124,21 +127,20 @@ bundle common va
...
-      "ncf_local_inputs_40" slist => splitstring(execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"ls -Path 'c:\Program Files\Rudder\var\ncf\local\40_it_ops_knowledge' -Name *.cf -Recurse -ErrorAction SilentlyContinue | % { Write-Host c:\Program Files\Rudder\var\ncf\local\40_it_ops_knowledge\$_ }\"", "useshell"), "\n", 10000);
-      "ncf_local_inputs_50" slist => splitstring(execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"ls -Path 'c:\Program Files\Rudder\var\ncf\local\50_techniques' -Name *.cf -Recurse -ErrorAction SilentlyContinue | % { Write-Host c:\Program Files\Rudder\var\ncf\local\50_techniques\$_ }\"", "useshell"), "\n", 10000);
-      "ncf_local_inputs_60" slist => splitstring(execresult("c:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -Command \"ls -Path 'c:\Program Files\Rudder\var\ncf\local\60_services' -Name *.cf -Recurse -ErrorAction SilentlyContinue | % { Write-Host c:\Program Files\Rudder\var\ncf\local\60_services\$_ }\"", "useshell"), "\n", 10000);
+      # We cannot merge both calls, as the paths have a space in them, and we need to quote them on windows
```
Comment (peckpeck) [inline techniques/system/common/1.0/promises.st]: Since you have a useshell, you can quote them on unix too

**Comment by ncharles** [comment]
Commit modified

**Comment by peckpeck** [comment]
Good
You can now merge by yourself using 
`rudder-dev merge https://github.com/Normation/rudder-techniques/pull/974`

**Comment by ncharles** [comment]
Commit modified

**Comment by peckpeck** [comment]
It's good for me, you can merge manually if you want. (or remove the can't merge label to try auromatically)

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**